hibernate-core/src/main/java/org/hibernate/query/spi/AbstractSelectionQuery.java

```diff
@@ -602,17 +597,6 @@ public R getSingleResultOrNull() {
 		}
 	}
 
-	protected static boolean hasLimit(SqmSelectStatement<?> sqm, MutableQueryOptions queryOptions) {
-		return queryOptions.hasLimit() || sqm.getFetch() != null || sqm.getOffset() != null;
-	}
-
-	protected static boolean hasAppliedGraph(MutableQueryOptions queryOptions) {
```

Comment:
Hey @gavinking,
Do you think it would be ok to add this particular method back to `AbstractSelectionQuery`? 
It was used in Hibernate Search, where there's a query extending `AbstractQuery`... if we do, it would be available to `AbstractSqmSelectionQuery` as well as to Search 🙈

Reply:
But .... the method is still there .... I just pulled it up the class hierarchy...

How did that break Search?

Reply:
Oh, no, wait, I see, that's a `static` method and it went _down_.

Ugh. It so does not belong on `AbstractSelectionQuery`. It only makes sense for SQM-based queries.

But sure, I guess you can add:

```java
@Deprecated(forRemoval=true)
protected static boolean hasAppliedGraph(MutableQueryOptions queryOptions) {
     return AbstractSqmSelectionQuery.hasAppliedGraph(queryOptions);
}
```

to `AbstractSelectionQuery`.

Probably your query class should be extending `AbstractSqmSelectionQuery`.

Reply:
mmm I think you've moved it to `AbstractSqmSelectionQuery`; but since Search is using the class from the spi package `org.hibernate.query.spiAbstractQuery extends AbstractSelectionQuery` we won't have access to it anymore and then it's protected + the `AbstractSqmSelectionQuery` is package-private so won't be able to get it through there 🙈

Reply:
> But sure, I guess you can add:

hmm, if it doesn't belong there, then it probably wouldn't make sense to add it back 😃 since we'd eventually need to get to it somehow. 

> Probably your query class should be extending AbstractSqmSelectionQuery.

I'm not sure about that one since it's in the internal package, would've been better if we could keep using something from the SPI, + `AbstractSqmSelectionQuery` is a package-private (at least for now). Let's ping @yrodiere to see what he'd say 🙈

Reply:
I mean the whole method is just this:

```java
	protected static boolean hasAppliedGraph(MutableQueryOptions queryOptions) {
		final AppliedGraph appliedGraph = queryOptions.getAppliedGraph();
		return appliedGraph != null && appliedGraph.getSemantic() != null;
	}
```

I'm pretty sure you could just copy/paste that.

Reply:
I think it's garbage, actually, I think it should be an error to have a null semantics.

If I change the code to throw when `graphSemantic` is null, no tests fail.

Reply:
Oh no no no, wait, wait ... I think it's actually checking the semantic when what it *should* be doing is checking that the _graph_ itself is non-null.

Reply:
> I mean the whole method is just this:
> 
> ```java
> 	protected static boolean hasAppliedGraph(MutableQueryOptions queryOptions) {
> 		final AppliedGraph appliedGraph = queryOptions.getAppliedGraph();
> 		return appliedGraph != null && appliedGraph.getSemantic() != null;
> 	}
> ```
> 
> I'm pretty sure you could just copy/paste that.

+1 that's the way forward in Hibernate Search 7.2 if we can't rely on those SPIs. @marko-bekhta can you please create a Jira and PR?

Now... this change will still prevent Hibernate Search 7.0/7.1 from working with Hibernate ORM 6.5.

For 7.0, well, too bad, but ultimately it's not our problem.

For 7.1 we can slip in a backport before the final release, and I think we should.